### DISCUSSION TITLE
Poetry install keeps hanging (Pending...)

### DISCUSSION THREAD

**Discussion by natv** [danielmiessler/Fabric#76] [Q&A]
Is there any trick to getting Fabric installed on Ubuntu 22.04.3 LTS ?  The poetry install keeps hanging. I even tried deleting the virtual env and trying all over again. This is a fresh Ubuntu install.

**I installed these packages initially:**

sudo apt install git python3-pip python3-poetry python3.10-venv build-essential libssl-dev libffi-dev zlib1g-dev

**Other steps I tried earlier were:**

- From a google search, some people had success with this command, but it doesn't work on my system:

**poetry config experimental.new-installer false**
(doesn’t work, says “Setting experimental.new-installer does not exist”

- **deleting the lock file**

- **disabling IPV6:**
sudo sysctl -w net.ipv6.conf.all.disable_ipv6=1
sudo sysctl -w net.ipv6.conf.default.disable_ipv6=1
sudo sysctl -w net.ipv6.conf.lo.disable_ipv6=1

- **Deleting the poetry cache**
cd ~/.cache/pypoetry/cache/
rm -rf artifacts/ cache/

- **poetry update** (also hung)

Then I completely deleted the virtual environment and started all over, and having the same issue:

**Here is what happens, it always seems to get stuck at the **macOS** keyring part:**

```

(venv) user@xxxxx:~/ai/fabric/client$ poetry install -vvv
Loading configuration file /home/user/.config/pypoetry/config.toml
Using virtualenv: /home/user/ai/fabric/client/venv
Installing dependencies from lock file

Finding the necessary packages for the current system

Package operations: 41 installs, 1 update, 0 removals, 6 skipped

  • Installing h11 (0.14.0): Pending...
[keyring.backend] Loading KWallet
[keyring.backend] Loading SecretService
[keyring.backend] Loading Windows
[keyring.backend] Loading chainer
[keyring.backend] Loading libsecret
[keyring.backend] Loading macOS


```

Thanks

**Comment by danielmiessler** [comment]
We're working on cleaning and optimizing. Thank you!

**Reply by natv** [reply]
the poetry install ended up working later in the day, must have been a problem with the repo side
